imap_processing/ena_maps/utils/naming.py

```diff
@@ -23,8 +26,350 @@ class MappableInstrumentShortName(Enum):
     GLOWS = "glx"
 
 
-# Ignore too many branches warning
-def build_l2_map_descriptor(  # noqa: PLR0912
+def get_instrument_descriptor(
+    instrument: MappableInstrumentShortName,
+    sensor: int | Literal["45", "90", "combined"] | str = "",
```

Comment:
Having both Literal and str for a type seems a bit odd. Do you need to accept generic strings too?

Reply:
I'll try to get rid of the str type here - it was an attempt to appease mypy, but there's probably a cleaner way

Reply:
I was able to remove this - it just required parsing 3 digit sensor numbers to ints and 2 digit ones to Literal of ["45", "90"]